keps/sig-apps/85-Graduate-PDB-to-Stable/README.md

```diff
@@ -0,0 +1,319 @@
+   * read the specified PodDisruptionBudget
+   * read status of the specified PodDisruptionBudget
+   * replace the specified PodDisruptionBudget
+   * replace status of the specified PodDisruptionBudget
+
+### Graduation Criteria
+
+Graduation to GA:
+- [x] Implement Mutable PDBs
+- [ ] Address performance issues
+- [ ] Pass conformance tests
+- [ ] Update documents to reflect the changes
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+* **How can this feature be enabled / disabled in a live cluster?**
+  - [x] Feature gate (also fill in values in `kep.yaml`)
+    - Feature gate name: PodDisruptionBudget
+    - Components depending on the feature gate: kube-controller-manager
+
+* **Does enabling the feature change any default behavior?**
+  No
+
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back
+  the enablement)?**
+  Yes
+
+* **What happens if we reenable the feature if it was previously rolled back?** 
+  No changes expected
+
+* **Are there any tests for feature enablement/disablement?**
+  Not applicable, given we're missing framework allowing switching feature-gates during e2e.
+
+### Rollout, Upgrade and Rollback Planning
+
+* **How can a rollout fail? Can it impact already running workloads?**
+  - If the leader election for the controller-manager fails, we can end up with two instances which will fight over the same PDB resources
+  - A panic in the disruption controller will crash the controller-manager
+
+* **What specific metrics should inform a rollback?**
+  - Unexpected controller-manager crashes
+
+* **Were upgrade and rollback tested? Was the upgrade->downgrade->upgrade path tested?**
+  Manual test is planned once the implementation is finished.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs, 
+fields of API types, flags, etc.?**
+  No
+
+### Monitoring Requirements
+
+* **How can an operator determine if the feature is in use by workloads?**
+  - Check number of PDBs in cluster with `etcd_object_counts{resource="poddisruptionbudgets.policy"}`
+  - Queue-related metrics to make sure the controller is functioning.
+
+* **What are the SLIs (Service Level Indicators) an operator can use to determine 
+the health of the service?**
+  - [x] Metrics
+    - Metric name: `workqueue_depth`
+    - Metric name: `workqueue_retries`
+    - Metric name: `workqueue_adds_total`
+    - Components exposing the metric: `kube-controller-manager`
+
+* **What are the reasonable SLOs (Service Level Objectives) for the above SLIs?**
+  99th percentile of `workqueue_depth` <= X per cluster-day.
+
+* **Are there any missing metrics that would be useful to have to improve observability 
+of this feature?**
+  No
+
+### Dependencies
+
+* **Does this feature depend on any specific services running in the cluster?**
+  No
+
+### Scalability
+
+* **Will enabling / using this feature result in any new API calls?** 
+  No
```

Comment:
If we talk about graduation to GA - I agree with this answer.

But if we talk about PDB feature itself - there are obviously new calls, e.g.
- disruption controller is watching for PDBs and pods (anything else?)
- disruption controller is updating PDB status based on what is happening with its pods

Especially the latter is potentially adding a non-negligible load (proportional to pod churn) so please describe what exactly is happening.


[I think it's the last chance to consciously make this audit...]

Reply:
Updated the KEP with a section that summarizes the calls to the API made by the disruption controller.